src/util/string.h

```diff
@@ -115,23 +118,34 @@ inline std::string padStringRight(std::string str, size_t len)
  *
  * @return If no end could be removed then "" is returned.
  */
-inline std::string removeStringEnd(const std::string &str,
+inline std::string_view removeStringEnd(std::string_view str,
 		const char *ends[])
 {
 	const char **p = ends;
 
 	for (; *p && (*p)[0] != '\0'; p++) {
-		std::string end = *p;
+		std::string_view end(*p);
 		if (str.size() < end.size())
 			continue;
 		if (str.compare(str.size() - end.size(), end.size(), end) == 0)
 			return str.substr(0, str.size() - end.size());
 	}
 
-	return "";
+	return std::string_view();
 }
 
 
+#define VARIANT_TEMPLATE typename T,
+#define VARIANT_CAST_TO std::basic_string_view<T>
```

Comment:
What's the benefit of these two?

Reply:
Nothing, I just made the macro more generic for no reason.